plugins/notifications-backend/migrations/20240302_numericSeverity.js

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+exports.up = async function up(knex) {
+  await knex.schema.alterTable('notification', table => {
+    table.tinyint('severity').notNullable().alter();
+    // we do not need to migrate data since there are not real deployments so far
```

Comment:
A bit concerned here as there might be real deployments already as the first version was published with `1.23.0`. I would do the migration just to be sure anyways. Is there a reason why this was changed to tinyint?

Reply:
Yep let's not assume that we can throw away data, even at this stage. I don't think this is a shortcut that we need to take and it prepares us for similar changes in the future.

I do wonder the same thing - how come the `tinyint`?

Reply:
It's outdated now. We didn't need any bigger number. The change was about representing severity as numbers - i.e. 0 for critical, 3 for low. Comparison `>=` is simpler. But we are back to text and handling it as strings.

Since we will probably keep textual counterparts, there will be a limited count of additions in the future.